src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy

```diff
@@ -274,60 +243,48 @@ class SentryPlugin implements Plugin<Project> {
 
                     def cli = getSentryCli(project)
 
-                    def persistIdsTaskName = "persistSentryProguardUuidsFor${variant.name.capitalize()}${variantOutput.name.capitalize()}"
-                    // create a task that persists our proguard uuid as android asset
-                    def persistIdsTask = project.tasks.create(
-                            name: persistIdsTaskName,
-                            type: Exec) {
-                        description "Write references to proguard UUIDs to the android assets."
-                        workingDir project.rootDir
-
-                        def propsFile = getPropsString(project, variant)
+                    def generateUuidTask = project.tasks.create(
+                            name: "generateSentryProguardUuid${variant.name.capitalize()}${variantOutput.name.capitalize()}",
+                            type: GenerateSentryProguardUuidTask) {
+                        outputDirectory.set(project.file("build/generated/assets/sentry/${variant.name}"))
```

Comment:
previously we're not appending `/sentry`, wondering if this wouldn't break reading the file from the APK

Reply:
No, because the path is hooked up so that it uses the output directory.

In fact, you should have unique output directories/files for each task since it's used in up-to-date checking and if they overlap you can get cache issues. This is the same pattern KAPT uses for example:
```
build/generated/kapt/release
```